runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,49 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview` / `start`, or `open` commands
+--- depending on the system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---
+---@return string returns the string output of the command. See |system()|.
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri)
+  local cmd
+
+  if vim.fn.has("wsl") then
+    cmd = "wslview"
+  elseif vim.fn.has("mac") then
+    cmd = 'open'
+  elseif vim.fn.has("win32") then
+    cmd = "start"
+  else -- Assume linux
+    cmd = 'xdg-open'
```

Comment:
There was also a good reason why I switched to `explorer` instead of `start` on Windows, but I can't remember why :)

Reply:
I cannot find much of a reason to prefer explorer. `start` might even be better since it can provide better filetype-handler errors where as `explorer` will just fail silently and open in the file explorer in a random default location. (I think)

Reply:
`start` is a "command" rather than an "executable". Using it instead of `explorer` here just errors out and does nothing

Reply:
curious

Reply:
If you're very keen on using `start` you could try to pass `cmd` + `path` as a joined string rather than a table I suppose. (`:help jobstart()` for full details)